conftest.py

```diff
@@ -90,6 +91,31 @@ def kill_all_test_spawned_processes():
     except Exception:
         print("Not all processes were killed after test")
 
+def get_hostlist():
+    global test_hostlist
+    if not test_hostlist:
+        if "COBALT_NODEFILE" in os.environ:
+            try:
+                with open(os.environ["COBALT_NODEFILE"], 'r') as nodefile:
+                    lines = nodefile.readlines()
+                    test_hostlist = [line.strip() for line in lines]
+            except:
+                return None
+        elif "COBALT_NODEFILE" in os.environ:
+            try:
+                with open(os.environ["PBS_NODEFILE"], 'r') as nodefile:
+                    lines = nodefile.readlines()
+                    test_hostlist = [line.strip() for line in lines]
+            except:
+                return None
+        elif "SLURM_JOB_NODELIST" in os.environ:
```

Comment:
Should we make generalized functions for these? I imagine this functionality might be a good candidate for ``smartsim.wlm``. Make a ticket? lets chat about this one.

Reply:
Yeah, totally makes sense, since some of our power users implemented their own versions of it.